apps/web-giddh/src/app/subscription/subscription.component.ts

```diff
@@ -311,11 +315,56 @@ export class SubscriptionComponent implements OnInit, OnDestroy {
             sessionId,
             sessionIndex
         };
-        this.store.dispatch(this.sessionAction.deleteSession(requestPayload));
+        const dialogRef = this.deleteConfirmationDialog(false);
+        dialogRef.afterClosed().pipe().subscribe(response => {
+            if (response === this.commonLocaleData?.app_yes) {
+                this.store.dispatch(this.sessionAction.deleteSession(requestPayload));
+            }
+        });
+    }
+
+    /**
+     * Deletes All sessions
+     *
+     * @memberof SubscriptionComponent
+     */
+    public clearAllSession(): void {
+        const dialogRef = this.deleteConfirmationDialog(true);
+        dialogRef.afterClosed().pipe().subscribe(response => {
+            if (response === this.commonLocaleData?.app_yes) {
+                this.store.dispatch(this.sessionAction.deleteAllSession());
+            }
+        });
     }
 
-    public clearAllSession() {
-        this.store.dispatch(this.sessionAction.deleteAllSession());
+    /**
+     *
+     *
+     * @param {boolean} allDeletes
+     * @return {*}  {*}
+     * @memberof SubscriptionComponent
+     */
+    public deleteConfirmationDialog(allDeletes: boolean): any {
+        const buttons: Array<ConfirmationModalButton> = [{
+            text: this.commonLocaleData?.app_yes,
+            color: 'primary'
```

Comment:
move to general service or approved by @dvCodeWorld to ignore

Reply:
Done